src/components/formik/Form/ScrollToErrorField/index.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useRef } from "react";
+
+import { useFormikContext } from "formik";
+
+import { getErrorFieldName } from "./utils";
+
+const ScrollToErrorField = ({ formRef }) => {
+  const { submitCount, isValid, errors } = useFormikContext();
+  const isValidatedReference = useRef(false);
+
+  useEffect(() => {
+    isValidatedReference.current = false;
+  }, [submitCount]);
+
+  useEffect(() => {
+    if (!formRef.current || isValidatedReference.current || isValid) return;
+    isValidatedReference.current = true;
+
+    const fieldErrorName = getErrorFieldName(errors);
+    if (!fieldErrorName) return;
+
+    const errorFormElement = formRef.current.querySelector(
+      `[name=${fieldErrorName}]`
+    );
+
+    errorFormElement?.scrollIntoView({
+      behavior: "smooth",
+      block: "center",
+    });
+  }, [submitCount, errors]);
```

Comment:
Wondering whether we can hook this into the `onSubmit` handler instead - since we are only triggering the scroll on submit. Always prefer using event handlers instead of `useEffects`

Reply:
@ajmaln If the form is invalid then `onSubmit`function will not be called. One other way to implement the scroll to error field is to use `onClick` handler of submit button. But then we must validate the form, which will add complications since validation is handled by validation schema. In the current implementation, we use `useEffect` with `errors` and `submitCount` from `formikContext` in the dependency array.